sklearn/utils/optimize.py

```diff
@@ -80,29 +139,36 @@ def newton_cg(func_grad_hess, func, grad, x0, args=(), eps=1e-4, tol=1e-4,
         ``max{|g_i | i = 1, ..., n} <= tol``
         where ``g_i`` is the i-th component of the gradient.
 
-    eps : float, optional
-        If fhess is approximated, use this value for the step size.
-
     maxiter : int
-        Number of iterations.
+        Number of Newton iterations.
+
+    maxinner : int
+        Number of CG iterations.
```

Comment:
I added maxinner because the inner loop could possibly be infinite (e.g., in case of incorrect gradient). Thanks to the convergence check (same as before), the inner loop should almost always stop before so there is not real point tuning maxinner. It should just not be too small.